### DISCUSSION TITLE
#+CAPTION: and ATTR_LATEX: function overwritten

### DISCUSSION THREAD

**Discussion by PiotrPanasiuk** [tbanel/orgaggregate#20] [General]
When one evaluate  `#+BEGIN: aggregate :table` caption and other functions which were added to table configuration are wiped out by aggregate function. Now my only solution is to move #+CAPTION: and ATTR_LATEX: directly above table and after evaluation move back to table. I also have some workaround:
`#+BEGIN: aggregate :table "SPECIAL" :cond (and (equal $9 "CMD") (not (equal $14 "ROB")) (not (equal $14 "TOD"))) :cols "$6;'Name' $14;'Command' count();'Counter'" :post (lambda (table) (append '(("#+CAPTION: List of commands\n" "#+ATTR_LATEX: :environment longtable :width \\linewidth :align |l|l|l| :font \\scriptsize") hline) table '(hline)))`
unfortunally as one can see it has some drawbacks.
 Is there possibility to have option which would just paste code before and after evaluation but inside #+BEGIN and #+END:?
Now I use workaround, but this is a bit cumbersome.

**Comment by tbanel** [comment]
Hi [PiotrPanasiuk](https://github.com/PiotrPanasiuk)

You would like something like that?

```
#+BEGIN: aggregate :table "SPECIAL" ...
#+CAPTION: List of commands
#+ATTR_LATEX: ...
|------+---------+---------|
| Name | Command | Counter |
|------+---------+---------|
|      |         |         |
|      |         |         |
#+END:
```

And have `#+BEGIN:` and `#+CAPTION:` kept in place when refreshing the aggregation?

This is a valid request.
Let me see what I can do.
Stay tuned.

**Comment by tbanel** [comment]
Fixed!
The `#+CAPTION:` and any other tag now survive refreshes.

The MELPA package will be updated within a few hours, or you can get the update right now in the GIT repo.

Thanks @PiotrPanasiuk for your contribution in the enhancement of OrgTbl Aggregate!

**Reply by PiotrPanasiuk** [reply]
Thank You very much, the change works great.
